packages/core/src/reporters/MutationTestReportCalculator.ts

```diff
@@ -0,0 +1,117 @@
+import * as path from 'path';
+import { MutantResult, mutationTestReportSchema, Reporter, MutantStatus } from '@stryker-mutator/api/report';
+import { StrykerOptions, Location, Position } from '@stryker-mutator/api/core';
+import { tokens, commonTokens } from '@stryker-mutator/api/plugin';
+import { coreTokens } from '../di';
+import { Logger } from '@stryker-mutator/api/logging';
+import InputFileCollection from '../input/InputFileCollection';
+import { normalizeWhitespaces } from '@stryker-mutator/util';
+
+export class MutationTestReportCalculator {
+
+  public static inject = tokens(coreTokens.reporter, commonTokens.options, coreTokens.inputFiles, commonTokens.logger);
+
+  constructor(
+    private readonly reporter: Required<Reporter>,
+    private readonly options: StrykerOptions,
+    private readonly files: InputFileCollection,
+    private readonly log: Logger
+  ) { }
+
+  public report(results: ReadonlyArray<MutantResult>) {
+    this.reporter.onMutationTestReportReady(this.mutationTestReport(results));
+  }
+
+  private mutationTestReport(results: ReadonlyArray<MutantResult>): mutationTestReportSchema.MutationTestResult {
+    return {
+      files: this.toFileResults(results),
+      schemaVersion: '1.0',
+      thresholds: this.options.thresholds
+    };
+  }
+
+  private toFileResults(results: ReadonlyArray<MutantResult>): mutationTestReportSchema.FileResultDictionary {
+    const files: mutationTestReportSchema.FileResultDictionary = Object.create(null);
+    results.forEach(mutantResult => {
+      const fileResult = files[mutantResult.sourceFilePath];
+      if (fileResult) {
+        fileResult.mutants.push(this.toMutantResult(mutantResult));
+      } else {
+        const sourceFile = this.files.files.find(file => file.name === mutantResult.sourceFilePath);
```

Comment:
You're inside of an anonymous function, referencing `this.files.files` while you have a variable in the method above called `files` and you have an instance variable called `files` as well. It's quite confusing which files is which and why you need to do `files.files`